app/DomainValueObjects/Location/Location.php

```diff
@@ -3,31 +3,54 @@
 namespace App\DomainValueObjects\Location;
 
 use App\DomainValueObjects\UUIDGenerator\UUID;
+use App\DomainValueObjects\Location\GeoLocation;
+use App\DomainValueObjects\Location\Address;
 
-//Exceptions
-//UUID
 use App\Exceptions\UUIDExceptions\GenerateUUID5Failed;
 use App\Exceptions\LocationExceptions\LocationNotDefined;
-
-
+use App\Exceptions\LocationExceptions\GeoLocationNotDefined;
+use App\Exceptions\LocationExceptions\AddressNotDefined;
 
 class Location
 {
-    private $uuid;
-    private $location = null;
+    private $uuid = null;
+    private $geoLocation = null;
+    private $address = null;
 
     public function __construct(
         UUID $uuid = null,
-        string $location = null //investigate data type
+        GeoLocation $geoLocation = null
     ) {
-        $this->uuid = $uuid->toString;
-        $this->location = $location;
+        $this->uuid = $uuid;
+        $this->geoLocation = $geoLocation;
         $this->validate();
     }
 
+    public static function withAddress(
+      UUID $uuid = null,
+      GeoLocation $geoLocation = null,
+      Address $address = null
+      )
+    {
+      $instance = new self($uuid, $geoLocation);
+      $instance->setAddress($address);
+      $instance->validateWithAddress();
+      return $instance;
+    }
+
+    protected function setAddress($address)
+    {
+      $this->address = $address;
```

Comment:
call validateWithAddress here instead of line 37, if we were to ever set a new address we would constantly have to do the line $instance->validateWithAddress();